firewall/incoming_blocker_interface.go

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (C) 2020 The "MysteriumNetwork/node" Authors.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package firewall
+
+import (
+	"net"
+)
+
+// IncomingTrafficBlocker interface neededs to be satisfied by any implementations which provide firewall capabilities, like iptables
+type IncomingTrafficBlocker interface {
```

Comment:
1. Interface name is IncomingTrafficBlocker but it also contains AllowIPAccess which doesn't block but allow access. Seems that it could be split into separate interfaces.
2. It contains Setup and Teardown which does unrelated things like DNS firewall setup which could itself be separate structure.

Could be separate interfaces.

```go
type IncomingTrafficBlocker interface {
    Block(network net.IPNet) (IncomingRuleRemove, error)
}

type IncomingTrafficAllower interface {
    Allow(network net.IPNet) (IncomingRuleRemove, error)
}
```

When Setup and Teardown could be moved to DNS related logic and firewall package will be decoupled from DNS rules setup.

Reply:
traffic blocker, neither blocks neither allows, it's interface to control how blocking rules are applied. If blocker suffix triggers you, i can rename it to IncomingTrafficFirewall :)

> When Setup and Teardown could be moved to DNS related logic and firewall package will be decoupled from DNS rules setup.

i can't add there. Setup needs to be initiated at very beginning of all system bootstrap

Reply:
_Firewall_ is a good name.

Reply:
Renamed